src/site/resources/js/redirect.js

```diff
@@ -0,0 +1,41 @@
+window.onload = () => {
+    const currentUrl = window.location.href;
+    if (currentUrl.endsWith("/checks/") || currentUrl.endsWith("/checks/index.html")) {
+        window.location.replace("../checks.html");
+    }
+    else if (document.title.startsWith("checkstyle – Redirecting to checks/")) {
+        const urlObj = new URL(currentUrl);
+        const pathSegments = urlObj.pathname.split("/");
+        const lastSegment = pathSegments[pathSegments.length - 1];
+        const extractedUrl =  `/${lastSegment}${urlObj.hash}`;
+        const pageTitles = new Map([
+            ["checkstyle – Redirecting to checks/coding.html", redirectConfigCodingUrls],
+            ["checkstyle – Redirecting to checks/header.html", redirectConfigHeaderUrls],
+        ])
+
+        const redirectFunc = pageTitles.get(document.title);
+        redirectFunc(extractedUrl);
+    }
+}
+
+const redirectConfigCodingUrls = (url) => {
+    switch (url) {
+        case "/config_coding.html":
+            window.location.replace("./checks/coding/");
+            break;
+        case "/config_coding.html#VariableDeclarationUsageDistance":
+            window.location.replace("./checks/coding/variabledeclarationusagedistance.html");
```

Comment:
Interesting idea:
If we do update like this for all links, it will be a lot, so js file will be with thousands lines, might not be quick to download 
:).

What if we just by code convert "#VariableDeclarationUsageDistance"
To "variabledeclarationusagedistance.html".
So at least js size will be small, so quick to download.

- [ ]

Reply:
Done. thanks a lot for the suggestion. Now our JS has reduced by a lot. I deleted the `redirect.js` and moved the code to `checkstyle.js` because our JS became so small.